airflow/api_fastapi/execution_api/app.py

```diff
@@ -34,10 +37,33 @@
 logger = structlog.get_logger(logger_name=__name__)
 
 
-@asynccontextmanager
-async def lifespan(app: FastAPI):
-    """Context manager for the lifespan of the FastAPI app. For now does nothing."""
+def _jwt_validator():
+    from airflow.configuration import conf
+
+    required_claims = frozenset(["aud", "exp", "iat"])
+
+    if issuer := conf.get("api_auth", "jwt_issuer", fallback=None):
+        required_claims = required_claims | {"iss"}
+    validator = JWTValidator(
+        required_claims=required_claims,
+        issuer=issuer,
+        leeway=conf.getint("api_auth", "jwt_leeway"),
+        audience=conf.get_mandatory_list_value("execution_api", "jwt_audience"),
+        **get_sig_validation_args(make_secret_key_if_needed=False),
+    )
+    return validator
+
+
+@svcs.fastapi.lifespan
+async def lifespan(app: FastAPI, registry: svcs.Registry):
     app.state.lifespan_called = True
+
+    # According to svcs's docs this shouldn't be needed, but something about SubApps is odd, and we need to
+    # record this here
+    app.state.svcs_registry = registry
+
+    # Create an app scoped validator, so that we don't have to fetch it every time
+    registry.register_value(JWTValidator, _jwt_validator(), ping=JWTValidator.status)
```

Comment:
I feel like introducing `svcs` just for caching `JWTValidator` might be a little bit overkill. (especially when we can achieve the same with fastapi and dependecy_override for tests) but I might be missing something.

(also not strongly against but I feel like this should be used if the usage was a little bit more intensive and spread outside of fastapi only)

Reply:
This is a low-impact way of introducing svcs to see what it feels like and if it's worth using larger (for instance using it for DB session management)